source/Cosmos.System/Graphics/Canvas.cs

```diff
@@ -230,6 +230,29 @@ public void DrawRectangle(Pen pen, int x, int y, int width, int height)
             DrawLine(pen, xc, yc, xd, yd);
         }
 
+        public void DrawFilledRectangle(Pen pen, int x_start, int y_start, int width, int height)
+        {
+            for (int i; i != width; i++)
+            {
+                DrawLine(pen, x_start, y_start, x_start + height, y_start);
+                y_start++;
+            }
+        }
+
+        public void DrawSquare(Pen pen, int x, int y, int sideslength)
```

Comment:
It doesn't make sense to have DrawSquare nor DrawFillesSquare, as it can be done using DrawRectangle and DrawFilledRectangle.